### PR TITLE
Add database watermarking for telemetry

### PR DESCRIPTION
**Pull request by dpeng817**
First pass at a watermarking scheme for the db. Hooks in to the run storage instead of setting up individual infrastructure. 
A few open questions:
1. Is there any additional data that we want to store outside of the id?
2. What should we do in the in-memory run storage case?
3. Are we ok hooking into the run storage like this?

### CODE REVIEW

File: python_modules/dagster/dagster/core/storage/runs/schema.py [hunk 1]
```diff
@@ -74,6 +74,12 @@
     db.Column("body", db.Text),
 )
 
+TelemetryTable = db.Table(
```
Comment (prha) [inline python_modules/dagster/dagster/core/storage/runs/schema.py:77]: only comment here is maybe this name is too specific.  Maybe this should be called instance or something like that?  

I'd also want to know if we'd want to track changes to the dagster.yaml...

Comment (dpeng817) [reply python_modules/dagster/dagster/core/storage/runs/schema.py:77]: yea we could hash it

**Comment by dpeng817** [comment]
Having some trouble getting the relation to actually be created in postgres... do I need to run a migration, or should that happen automagically?

**Comment by prha** [comment]
This kind of depends on the actual storage implementation.  If you look at postgres run_storage, for example, there's a flag `should_autocreate_tables` and there's a guard checking for existing tables:

```
        if self.should_autocreate_tables and "runs" not in table_names:
            retry_pg_creation_fn(self._init_db)
            self.build_missing_indexes()
```

**Review by prha — CHANGES_REQUESTED** [review]
rename db_watermark / telemetry_id in favor of instance_id / run_storage_id / etc.

**Pull request merged**